tcp_check/datadog_checks/tcp_check/tcp_check.py

```diff
@@ -110,47 +117,60 @@ def check(self, _):
-                self.report_as_service_check(
-                    AgentCheck.CRITICAL, "{}. Connection failed after {} ms".format(str(e), length)
-                )
 
-            if self.socket_type == socket.AF_INET:
-                self.log.debug("Will attempt to re-resolve IP for %s:%d on next run", self.host, self.port)
-                self._addr = None
+        for addr in self.addrs:
+            try:
+                response_time = self.connect(addr)
+                self.log.debug("%s:%d is UP (%s)", self.host, self.port, addr)
+                self.report_as_service_check(AgentCheck.OK, addr, 'UP')
+                if self.collect_response_time:
+                    self.gauge(
+                        'network.tcp.response_time',
+                        response_time,
+                        tags=self.tags + ['address:{}'.format(addr)],
+                    )
+            except Exception as e:
+                length = int((get_precise_time() - start) * 1000)
+                if isinstance(e, socket.error) and "timed out" in str(e):
+                    # The connection timed out because it took more time than the system tcp stack allows
+                    self.log.warning(
+                        'The connection timed out because it took more time '
+                        'than the system tcp stack allows. You might want to '
+                        'change this setting to allow longer timeouts'
+                    )
+                    self.log.info("System tcp timeout. Assuming that the checked system is down")
+                    self.report_as_service_check(
+                        AgentCheck.CRITICAL,
+                        addr,
+                        """Socket error: {}.
+                    The connection timed out after {} ms because it took more time than the system tcp stack allows.
+                    You might want to change this setting to allow longer timeouts""".format(
+                            str(e), length
+                        ),
+                    )
+                else:
+                    self.log.info(
+                        "%s:%d is DOWN (%s) (%s). Connection failed after %d ms",
+                        self.host,
+                        self.port,
+                        addr,
+                        str(e),
+                        length,
+                    )
+                    self.report_as_service_check(
+                        AgentCheck.CRITICAL, addr, "{}. Connection failed after {} ms".format(str(e), length)
+                    )
+
+                if self.socket_type == socket.AF_INET:
+                    self.log.debug("Will attempt to re-resolve IP for %s:%d on next run", self.host, self.port)
+                    self._addrs = None
 
-    def report_as_service_check(self, status, msg=None):
+    def report_as_service_check(self, status, addr, msg=None):
         if status is AgentCheck.OK:
```

Comment:
Can you also leave a note somewhere (maybe the PR title) to signify we're also adding the `address` tag everywhere?